query/src/pipelines/new/processors/transforms/hash_join/chaining_hash_table.rs

```diff
@@ -469,11 +468,32 @@ impl ChainingHashTable {
                         {
                             let keys_ref = KeysRef::create(key.as_ptr() as usize, key.len());
                             let probe_result_ptr = table.hash_table.find_key(&keys_ref);
-                            if probe_result_ptr.is_none() {
-                                // No matched row for current probe row
-                                continue;
+                            match (probe_result_ptr, &self.other_predicate) {
```

Comment:
Should we also modify `result_blocks` ?

Reply:
> Should we also modify `result_blocks` ?

We don't actually need to, this is the Semi/Anti join-specific behavior.